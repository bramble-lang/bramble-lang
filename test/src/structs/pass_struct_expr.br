
fn my_main() -> i64 {
    let y:i64 := test(10, MyStruct{x: 1111, y: 2222, z: 3333});
    root::std::io::writei64ln(y);
    
    return;
}

struct MyStruct {
    x: i64,
    y: i64,
    z: i64,
}

fn test(x: i64, t: MyStruct) -> i64 {
    root::std::io::writei64ln(x);
    root::std::io::writei64ln(t.x);
    root::std::io::writei64ln(t.y);
    root::std::io::writei64ln(t.z);

    return x * (t.x + t.y + t.z);
}