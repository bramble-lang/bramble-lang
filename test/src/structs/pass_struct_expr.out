Error: Semantic: L3: Cannot pass struct expression as function parameter (future feature)
